ff/src/fields/mod.rs

```diff
@@ -115,7 +115,8 @@ pub trait Field:
     + From<bool>
 {
     type BasePrimeField: PrimeField;
-
+    type BasePrimeFieldArr: AsRef<[Self::BasePrimeField]>;
```

Comment:
Maybe we change this to `type BasePrimeFieldIter: Iterator<Item = Self::BasePrimeField>;`?